packages/charts/src/chart_types/metric/renderer/dom/text.tsx

```diff
@@ -305,7 +313,33 @@ export const MetricText: React.FunctionComponent<{
   );
 };
 
-function splitNumericSuffixPrefix(text: string): { emphasis: 'normal' | 'small'; text: string }[] {
+function getTextParts(datum: MetricDatum, style: MetricStyle): TextParts[] {
+  if (isMetricWStringArrayValues(datum)) return [{ emphasis: 'normal', text: `[${datum.value.join(', ')}]` }];
+
+  if (isMetricWNumberArrayValues(datum)) {
+    const numericValueParts = datum.value.reduce<TextParts[]>(
+      (acc, value, i, { length }) => {
+        const parts: TextParts[] = isFiniteNumber(value)
+          ? splitNumericSuffixPrefix(datum.valueFormatter(value))
+          : [{ emphasis: 'normal', text: style.nonFiniteText }];
+        if (i < length - 1) {
+          parts.push({ emphasis: 'normal', text: ', ' });
+        }
+        return [...acc, ...parts];
+      },
+      [{ emphasis: 'normal', text: '[' }],
+    );
+    return [...numericValueParts, { emphasis: 'normal', text: ']' }];
+  }
+
+  return isMetricWNumber(datum)
+    ? isFiniteNumber(datum.value)
+      ? splitNumericSuffixPrefix(datum.valueFormatter(datum.value))
+      : [{ emphasis: 'normal', text: style.nonFiniteText }]
+    : [{ emphasis: 'normal', text: datum.value }];
+}
```

Comment:
You can avoid writing the same operation if you transform your `isMetricWNumber` into a single array element.

```suggestion
 
    const numericValueParts = (isMetricWNumberArrayValues(datum) ? datum.value : [datum.value]).reduce<TextParts[]>(
      (acc, value, i, { length }) => {
        const parts: TextParts[] = isFiniteNumber(value)
          ? splitNumericSuffixPrefix(datum.valueFormatter(value))
          : [{ emphasis: 'normal', text: style.nonFiniteText }];
        if (i < length - 1) {
          parts.push({ emphasis: 'normal', text: ', ' });
        }
        return [...acc, ...parts];
      },
      [{ emphasis: 'normal', text: '[' }],
    );
    return [...numericValueParts, { emphasis: 'normal', text: ']' }];
}
```

Reply:
No this code would treat everything like an array.

I was trying to keep it simple and easier to read but I combined them all in a66c6b175f6a01069c58ad52980d836cede51d42. There's more type checking here with all unions also checking for `NaN`. But this is as compact as I can get it. WDYT?

Reply:
>this code would treat everything like an array.

correct, this to limit the code duplication, because in both cases you are checking for finiteNumber, splitting, etc.
All fine for me, except that you can move the `valueFormatter` call outside the reduce to avoid calling it everytime.

Reply:
Done f65840a